lib/rubygems/specification.rb

```diff
@@ -1530,7 +1530,7 @@ def authors
   # a full path.
 
   def bin_dir
-    @bin_dir ||= File.join gem_dir, bindir # TODO: this is unfortunate
```

Comment:
Do you think that TODO it related to the method NOTE and possible confusion?

Reply:
I thik it is!, thanks for pointing that out @simi 

Maybe rename `bin_dir` to `bin_dir_path` ?

Reply:
Uhh, I think changing public specification method signature is not good idea. Anyway `NOTE` is probably enough in here. There's no need for this TODO.

Reply:
I was not clear... Yeah, I meant as suggestion for the future